rewrite-java/src/main/java/org/openrewrite/java/tree/Pattern.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.tree;
+
+public interface Pattern extends J {
```

Comment:
Since the only place we currently refer to this type is the `implements` clause of `DeconstructorPattern` I am thinking we might as well remove it for now and add it once we need it (e.g. as a parameter or return type somewhere).